### PR TITLE
[Feature] Create a custom link browser tutorial

### PR DESCRIPTION
**Pull request by linawolf**
This PR is only about the tutorial to keep changes simple enough for review

Next steps: 
* Overhaul an unify the APIsection of the link handler. Link from and to link handler API once that section is cleared up.

References:#2298

### CODE REVIEW

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:63]:
```suggestion
For technical reasons, not all dependencies needed by the backend link handler can
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:68]: What does the AbstractLinkBrowserController and why it is set?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:71]: Why is the PageRenderer needed?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:80]:
```suggestion
    and its call is an implementation detail that might be
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:74]: Why do we have two views? The one here and the BackendViewFactory view?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:95]:
```suggestion
When the button in the rendered form is clicked to set a link, a custom
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_CustomLinkHandlerJavaScript.rst.txt
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_CustomLinkHandlerJavaScript.rst.txt:8]: Do we need this full comment?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:105]:
```suggestion
If not done yet, the JavaScript has to be registered in the file
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:113]:
```suggestion
:php:`backend` has to be added as dependency. See also
```

More, what has to be done instead of where. The where is obvious from the snippet.

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:112]:
```suggestion
As our JavaScript class depends on classes provided by the backend system extension,
```

This way, it is more obvious that this is provided by TYPO3 itself than some extension.

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:120]:
```suggestion
user edits an existing link in the link browser. All backend link handlers will
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:121]:
```suggestion
be called and can decide if they can handle that link. If so, they should store
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:122]:
```suggestion
the provided information to be used in rendering (for example, to fill an input
```

Additionally: what rendering?

And I don't understand the example "to fill an input field with the old value". Why should I want to do that? And which field?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:130]: "... is used as preview for the link in the backend ..."

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:131]:
```suggestion
what the link will look like in the backend, for example, in the upper part of
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:167]: What does this mean, a parsed version?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:185]:
```suggestion
configuration and data from the typolink function. If the link can be rendered,
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:186]:
```suggestion
it returns a new :php:`\TYPO3\CMS\Frontend\Typolink\LinkResultInterface`. The
```

Nope, is does not return an interface, it returns a class implementing the interface.

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:191]:
```suggestion
:php:`\TYPO3\CMS\Frontend\Typolink\UnableToLinkException`.
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:190]:
```suggestion
If the link cannot be built it should throw a
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerFormatCurrentUrl.rst.txt
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerFormatCurrentUrl.rst.txt:11]:
```suggestion
           return 'https://github.com/' . $this->configuration['project'] . '/'
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt:28]: What for is the BackendViewFactory?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GithubLinkBuilder.rst.txt
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GithubLinkBuilder.rst.txt:19]: Why linkDetails by reference?

**Review by brotkrueml — CHANGES_REQUESTED** [review]
No merge into 11.5?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:9]: We only create the tab / tab content, not the complete window

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:17]: Not sure how to make a positive sentence about this. You need a custom link handler, when the RecordLinkBrowser is not sufficient. Thats why I added the link and the tip

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:38]: Again we do only create the content of the tab

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:131]:
```suggestion
The function :php:`LinkHandlerInterface::formatCurrentUrl()` is used as preview 
for the link in the backend, for example in the upper part of
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_CustomLinkHandlerJavaScript.rst.txt
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_CustomLinkHandlerJavaScript.rst.txt:8]: No that is what the todo is for. The codesnippet generator has to be changed so it can also filter these comments from JavaScript

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerFormatCurrentUrl.rst.txt
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerFormatCurrentUrl.rst.txt:11]: We would need to change that in examples...

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt:28]: Might be a relict, however we need to anylize and fix this in examples not here

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GithubLinkBuilder.rst.txt
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GithubLinkBuilder.rst.txt:19]: The method signature is defined like that in AbstractTypolinkBuilder

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:30]:
```suggestion
automatically.

**Step by step to a custom link browser**

..  contents::
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:73]: to register the JavaScript, see render method. Shall we put this information here as well?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GithubLinkBuilder.rst.txt
Comment (brotkrueml) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GithubLinkBuilder.rst.txt:19]: Puuh, old and bad code...

File: Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt
Comment (brotkrueml) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt:28]: Yup, that should be done there before merging here.

**Comment by linawolf** [comment]
> No merge into 11.5?

At least no automatic one. ES6 does not work in v11 yet and I think there have been some other API changes as well

**Comment by linawolf** [comment]
On hold until necessary changes have been merged into EXT:examples

**Comment by brotkrueml** [comment]
> > No merge into 11.5?
> 
> At least no automatic one. ES6 does not work in v11 yet and I think there have been some other API changes as well

Okay, I am fine with that.

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:9]: This is not what I meant. The sentence above focused on creating a tab: "In this tutorial we create a custom link browser tab ...". Nope, we create a custom link browser, nor only a tab in backend.

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:17]: Yup, just a remark as you know the docs about link handler better than me. Would be also better in another PR and possibly another doc.

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [reply Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:9]: How would you phrase that?

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:104]:
```suggestion
:php:`AbstractLinkBrowserController $linkBrowser`
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:112]: I suspect, string[] is not correct. Valid page TSconfig may be:

[
    'some_key' => 'some_value'
]

which is equivalent to `array<string, string>`, but also `array<string, array<string, string>>` is valid.

But string[] means:

[
    'some_value'
]

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:113]:
```suggestion
    Contains the page TSconfig configuration as array of the link browser tab
```

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (brotkrueml) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:106]:
```suggestion
    configuration information of the complete link browser window.
```

**Review by brotkrueml — COMMENTED** [review]
Sorry, some minor glitches found.

File: Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst
Comment (linawolf) [inline Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst:112]:
```suggestion
:php:`array $configuration`
```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**
